File: ABC/D/256/D.rb
query = Array.new { Array.new(2) }
gets.chomp.to_i.times do
    l, r = gets.chomp.split.map(&:to_i)
    query << [l, 0]
    query << [r, 1]
end
query.sort!
cnt = 0
query.each do |vec|
    if vec[-1] == 0 then
        print "#{vec[0]} " if cnt == 0
        cnt += 1
    else
        cnt -= 1
        p vec[0] if cnt == 0
    end
end

File: ABC/D/256/D.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main(){
    int n;
    vector<pair<int, int>> query;
    cin >> n;
    while(n--){
        int l, r;
        cin >> l >> r;
        query.push_back(make_pair(l, false));
        query.push_back(make_pair(r, true));
    }
    sort(query.begin(), query.end());
    int cnt=0;
    for(auto iter: query){
        if(!iter.second){
            if(cnt == 0) cout << iter.first << ' ';
            cnt++;
        }
        else{
            cnt--;
            if(cnt == 0) cout << iter.first << endl;
        }
    }
}